test/gitops_test.go

```diff
@@ -0,0 +1,128 @@
+	allowAction = crdv1alpha1.RuleActionAllow
+	np1         = &networkingv1.NetworkPolicy{
+		TypeMeta:   metav1.TypeMeta{Kind: "NetworkPolicy", APIVersion: "networking.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{Namespace: "nsA", Name: "npA", UID: "uidA"},
+		Spec: networkingv1.NetworkPolicySpec{
+			PodSelector: metav1.LabelSelector{},
+			PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeIngress},
+			Ingress:     []networkingv1.NetworkPolicyIngressRule{{}},
+		},
+	}
+	anp1 = &crdv1alpha1.NetworkPolicy{
+		TypeMeta:   metav1.TypeMeta{Kind: "NetworkPolicy", APIVersion: "crd.antrea.io/v1alpha1"},
+		ObjectMeta: metav1.ObjectMeta{Namespace: "nsA", Name: "anpA", UID: "uidC"},
+		Spec: crdv1alpha1.NetworkPolicySpec{
+			AppliedTo: []crdv1alpha1.NetworkPolicyPeer{
+				{PodSelector: &selectorA},
+			},
+			Priority: p10,
+			Ingress: []crdv1alpha1.Rule{
+				{
+					Ports: []crdv1alpha1.NetworkPolicyPort{
+						{
+							Port: &int80,
+						},
+					},
+					From: []crdv1alpha1.NetworkPolicyPeer{
+						{
+							PodSelector:       &selectorB,
+							NamespaceSelector: &selectorC,
+						},
+					},
+					Action: &allowAction,
+				},
+			},
+			Egress: []crdv1alpha1.Rule{
+				{
+					Ports: []crdv1alpha1.NetworkPolicyPort{
+						{
+							Port: &int81,
+						},
+					},
+					To: []crdv1alpha1.NetworkPolicyPeer{
+						{
+							PodSelector:       &selectorB,
+							NamespaceSelector: &selectorC,
+						},
+					},
+					Action: &allowAction,
+				},
+			},
+		},
+	}
+)
+
+func TestHandleEventList(t *testing.T) {
+	fakeClient := NewClient(np1, anp1)
+	k8s := &gitops.K8sClient{
+		Client: fakeClient,
+	}
+
+	jsonstring, err := ioutil.ReadFile("./files/correct-audit-log.txt")
+	if err != nil {
+		t.Errorf("Error (TestHandleEventList): could not read correct-audit-log file")
+	}
+	cr, err := gitops.SetupRepo(k8s, gitops.StorageModeInMemory, directory)
+	if err != nil {
+		t.Errorf("Error (TestHandleEventList): could not set up repo")
+	}
+
+	err = cr.HandleEventList(jsonstring)
+	if err != nil {
+		t.Errorf("Error (TestHandleEventList): could not handle correct audit event list")
+	}
+
+	cr.RollbackMode = true
+	if cr.HandleEventList(jsonstring) == nil {
+		t.Errorf("Error (TestHandleEventList): could not return correct error with roll back")
+	}
+	cr.RollbackMode = false
+
+	for i := 1; i < 4; i++ {
+		filename := fmt.Sprintf("%s%d%s", "./files/incorrect-audit-log-", i, ".txt")
+		jsonstring, err := ioutil.ReadFile(filename)
+		if err != nil {
+			cr.HandleEventList(jsonstring)
+			if err == nil {
+				t.Errorf("Error (TestHandleEventList): could not return error on erroneous json file")
+			}
+		}
+	}
+}
+
+func NewClient(objects ...runtime.Object) client.WithWatch {
+	scheme := runtime.NewScheme()
+	gitops.RegisterTypes(scheme)
+	clientBuilder := fake.NewClientBuilder()
+	clientBuilder.WithRuntimeObjects(objects...)
+	clientBuilder.WithScheme(scheme)
+	client := clientBuilder.Build()
+	return client
+}
```

Comment:
I separated the tagging/rollback tests into other files (see #7), so maybe rename this to audit_handler_test.go or something like that

Reply:
Should each .go file will have an according _test.go file? That might be a clean way to organize.